### DISCUSSION TITLE
Date on Ecliptic of Date diverges from Date and Time dialog in the past

### DISCUSSION THREAD

**Discussion by 101610** [Stellarium/stellarium#3969] [Q&A] [answered]
If I set the Date and Time dialog on 2024/11/4, the Ecliptic of date shows the sun on or near XI.4.
In the past, the two dates seem to diverge.

If I set the Date and Time dialog on -6000/11/4, the Ecliptic of date shows the sun on or near XI.7.
I expected XI.4 (as was the case in 2024) rather than XI.7.
Why do the two dates diverge?

24.3 with DE431 on Windows 11

![Ecliptic_of_date2](https://github.com/user-attachments/assets/73f99110-60ab-4de3-b92a-5b429565259c)

**Comment by gzotti** [comment]
User Guide, Appendix F; Wikipedia on Gregorian Calendar Reform, ...

**Comment by 101610** [comment]
I understand different calendars give different numbers and names for the same day.
I do not understand which calendar I see on the ecliptic in Stellarium.

The manual (13.9.1 and F.8) explains about calendar dates in "the main program":
dates before the Gregorian calendar from 1582 are given in the Julian calendar.
I assume dates in "the main program" means the Date and Time window (not dates on the ecliptic of date, they are different).
I also understand (13.9.2) the Calendar plugin shows a proleptic Gregorian calendar, among others.

So, 
(a) -6000/11/4 (November 4) in the Date and Time window = Julian calendar
The Julian calendar in the Calendar plugin shows the same month and day.

(b) Same day in the Calendar plugin is:
-6000/9/18 (September 18) = proleptic Gregorian calendar

(c) Same day on the ecliptic of date is:
XI.7 (November 7) = which calendar is this?

The proleptic Gregorian calendar (b) counts 365.2425 days (13.9.2). Counting back from 1582, by -6000 this calendar would differ by about 2.3 days from an "exact" solar calendar of 365.2422 days (7,582 years x 0.0003 days, or about 3 days in 10,000 years).
The day on the ecliptic of date (c) differs about 50 days from the proleptic Gregorian calendar (b) and 3 days from the Julian calendar (a). What explains the difference?

**Reply by gzotti** [reply, marked as answer]
OK, this deserves another look into the code. The scale along the ecliptic should be Julian, and it is rebuilt when the year changes. But who knows what calendar people used in -6000... Just give solar longitudes for reference.

**Reply by 101610** [reply]
Thanks for looking into this.

> But who knows what calendar people used in -6000... Just give solar longitudes for reference.

Right.
In archaeoastronomy, I often find it helpful to see day numbers rather than a calendar that in some cases was not used at that time and place.

Can Stellarium show an ecliptic with day numbers (1, 2, 3...)?

Or is that possible with the ArchaeoLines Plug-in? We can set custom lines, very helpful. I would find it useful to be able to set a custom ecliptic showing simple day numbering (starting from the vernal equinox or preferably a longitude of choice) and/or a calendar of choice.

**Reply by gzotti** [reply]
The ecliptic can show a degree scale which is usually close enough to a day scale. 
A large part of the calendar offset BTW is caused by DeltaT which in this time brings in an uncertainty of days. Again, for 6000BC, you'd better speak in terms of solar longitude and explain what it means as approximate date for readers unfamiliar with the astronomical roots of their own calendar. Maybe I should just suppress the calendar scale for 500BC and earlier.

**Reply by alex-w** [reply]
> Maybe I should just suppress the calendar scale for 500BC and earlier.

Hmm... probably start point should be the date of starting Julian calendar.

**Reply by gzotti** [reply]
That's the brutal solution. But then I should also implement the messed-up execution by the Roman pontifices (leaping every 3 years) until Augustus fixed it. No, many historians give dates in the Julian calendar for Babylonian and Egyptian history, some of them linked to astronomical events like eclipses (from which DeltaT is derived for these times). However, most really early dates just mention years, not specific celestial events where this would matter. I might have to take another look into computing the day scale and its DeltaT connection. 

Would it help to allow labeling the Ecliptic degree scale not only 0..360° but in the antique/medieval tradition of 12 signs of 0...30° each?

A truly authentic program should give "day 19 in the 3rd month in the 18th year of King XY" depending on your current location, and no reference to our calendar :-) However, few calendars and chronicles are complete enough to allow that.

**Reply by alex-w** [reply]
> Would it help to allow labeling the Ecliptic degree scale not only 0..360° but in the antique/medieval tradition of 12 signs of 0...30° each?

Just an additional sign every 30 degree near numbers of degrees

> A truly authentic program should give "day 19 in the 3rd month in the 18th year of King XY" depending on your current location, and no reference to our calendar :-) However, few calendars and chronicles are complete enough to allow that.

Maybe just a number of day in year?

**Reply by gzotti** [reply]
Then we must discuss when the year begins in whose chronology... Jan1, March1, or Vernal equinox? This is why I recommend just giving solar longitudes. As said, exact numbers linked to solar longitude expectations seem to fall apart with DeltaT exceeding a day. But I need to re-check the code of ecliptic date mark generation when it comes to the Mesolithic as it does not fulfill its purpose of showing Solar position. This is the basic message of this thread. Certainly not this or next week, though, and don't hold your breath for a solution in 24.4.

**Reply by alex-w** [reply]
> Then we must discuss when the year begins in whose chronology... Jan1, March1, or Vernal equinox?

It may be user defined option

**Reply by gzotti** [reply]
I think the ~50 users who work in this field and period should be happy with ecliptical longitudes and Historians' common Julian calendar. But yes, the latter may need another look. The date marks were inspired by those in real planetaria, which are also never exact (giving average Greg. dates for present-day explanations; our marks are recomputed at year change.).

**Comment by 101610** [comment]
> Then we must discuss when the year begins in whose chronology... Jan1, March1, or Vernal equinox? 

If you make a day scale at all (in the Archaeolines plugin?), one reasonable default for day 1 is an equinox. But it would be great indeed if you can customize day 1 to a longitude of choice so that you can also set the other equinox, Jan1, some heliacal rising or any day that interests you as day 1.

> Maybe I should just suppress the calendar scale for 500BC and earlier.

Either way, I would not suppress the ecliptic calendar scale. It's an optional marking anyway and certainly useful for reference even if it's not always ideal.

The original question has been answered. Thanks.
